### PR TITLE
remove casttype

### PR DESCRIPTION
**Pull request by likhita-809**

### CODE REVIEW

File: x/group/server/msg_server.go [hunk 1]
```diff
@@ -191,7 +192,10 @@ func (s serverImpl) UpdateGroupAdmin(ctx types.Context, req *group.MsgUpdateGrou
 	action := func(g *group.GroupInfo) error {
 		g.Admin = req.NewAdmin
 		g.Version++
-		return s.groupTable.Save(ctx, g.GroupId.Bytes(), g)
+
+		groupID := group.ID(g.GroupId).Bytes()
+
```
Comment (blushi) [inline x/group/server/msg_server.go:197]:
```suggestion
```

File: x/group/server/query_server.go [hunk 2]
```diff
@@ -42,7 +43,8 @@ func (s serverImpl) getGroupAccountInfo(ctx types.Context, accountAddress sdk.Ac
 }
 
 func (s serverImpl) GroupMembers(ctx types.Context, request *group.QueryGroupMembersRequest) (*group.QueryGroupMembersResponse, error) {
-	it, err := s.getGroupMembers(ctx, request.GroupId, request.Pagination)
+	groupID := group.ID(request.GroupId)
+	it, err := s.getGroupMembers(ctx, groupID, request.Pagination)
```
Comment (blushi) [inline x/group/server/query_server.go:47]: I guess we could also just use `uint64` in the place of `group.ID` as param of `getGroupMembers`

File: x/group/server/query_server.go [hunk 3]
```diff
@@ -90,7 +92,8 @@ func (s serverImpl) getGroupsByAdmin(ctx types.Context, admin sdk.AccAddress, pa
 }
 
 func (s serverImpl) GroupAccountsByGroup(ctx types.Context, request *group.QueryGroupAccountsByGroupRequest) (*group.QueryGroupAccountsByGroupResponse, error) {
-	it, err := s.getGroupAccountsByGroup(ctx, request.GroupId, request.Pagination)
+	groupID := group.ID(request.GroupId)
+	it, err := s.getGroupAccountsByGroup(ctx, groupID, request.Pagination)
```
Comment (blushi) [inline x/group/server/query_server.go:96]: ditto use `uint64` in the place of `group.ID` in `getGroupAccountsByGroup`

File: x/group/types.go [hunk 4]
```diff
@@ -286,9 +286,14 @@ const MaxMetadataLength = 255
...
 func (g GroupInfo) ValidateBasic() error {
-	if g.GroupId.Empty() {
+	if g.GroupId == 0 {
 		return sdkerrors.Wrap(ErrEmpty, "group")
 	}
+	/*
+		if g.GroupId.Empty() {
+			return sdkerrors.Wrap(ErrEmpty, "group")
+		}
+	*/
```
Comment (blushi) [inline x/group/types.go:296]: can we remove that?

File: x/group/types.go [hunk 5]
```diff
@@ -307,9 +312,14 @@ func (g GroupInfo) ValidateBasic() error {
...
 func (g GroupMember) ValidateBasic() error {
-	if g.GroupId.Empty() {
+	if g.GroupId == 0 {
 		return sdkerrors.Wrap(ErrEmpty, "group")
 	}
+	/*
+		if g.GroupId.Empty() {
+			return sdkerrors.Wrap(ErrEmpty, "group")
+		}
+	*/
```
Comment (blushi) [inline x/group/types.go:322]: ditto

**Review by blushi — CHANGES_REQUESTED** [review]
Thanks @likhita-809, looks good overall, just left minor comments! Also you might wanna resolve the merge conflicts.

**Review by robert-zaremba — APPROVED** [review]
I like it - code easier to read! Please look at Marie comments.

**Review by blushi — APPROVED** [review]

**Comment by blushi** [comment]
thanks @likhita-809, I'll merge this!
Please make sure to resolve the conversations from the review once you've addressed them ;)

**Pull request merged**